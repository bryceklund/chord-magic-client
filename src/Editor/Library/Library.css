html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed, 
figure, figcaption, footer, header, hgroup, 
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 100%;
	vertical-align: baseline;
}

article, aside, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section {
	display: block;
}

body {
	line-height: 1;
}

ol, ul {
	list-style: none;
}

blockquote, q {
	quotes: none;
}

blockquote:before, blockquote:after,
q:before, q:after {
	content: '';
	content: none;
}

table {
	border-collapse: collapse;
	border-spacing: 0;
}

* {
  font-family: Arial, Helvetica, sans-serif;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

body {
  margin: 0;
  padding-bottom: 50px;
}

nav {
  width: 100%;
  height: 45px;
  position: absolute;
  top: 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
  background-color: #27CCBE;
}

.timeline {
  margin-top: 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}

.global_view {
  margin-top: 5px;
  margin-bottom: 5px;
}

.play_stop_buttons {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.player_timeline {
  width: 500px;
  height: 180px;
  background-color: black;
  display: flex;
  flex-direction: row;
  margin-top: 5px;
  margin-bottom: 5px;
}

.chord_label {
  font-size: 40px;
  position: absolute;
  text-align: center;
}

.move_chord_buttons {
  display: flex;
  height: 35px;
  position: absolute;
  bottom: 0;
  width: 100%;
}

.move_chord_buttons > button {
  display: flex;
  align-items: center;
  justify-content: center;
}

.scrubber_buttons {
  width: 500px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.slide_container {
  text-align: center;
}

.library_container {
  margin-top: 30px;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
}

.library_settings {
  display: flex;
  flex-direction: row-reverse;
  width: 700px;
  height: 300px;
  background: #27CCBE;
  border-radius: 15px;
}

.wave_icon {
  width: 50px;
}

[type=radio] { 
  position: absolute;
  opacity: 0;
  width: 0;
  height: 0;
}

/* IMAGE STYLES */
[type=radio] + img {
  cursor: default;
}

.wave_icon:hover {
  outline: 1px solid lightgray;
}

/* CHECKED STYLES */
[type=radio]:checked + img {
  outline: 1px solid #000000;
}

.wave_icons_container {
  display: flex;
  flex-direction: column;
  margin-top: 5px;
}

.instrument_container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.library {
  width: 75%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}

.scales {
  list-style: none;
  padding-left: 0;
  columns: 3;
}

.scale a, .scale a:visited {
  text-decoration: none;
  color: black;
}

.scale, .note {
  font-size: 50px;
  width: 150px;
  text-align: center;
  margin-bottom: 10px;
  cursor: copy;
}

.note {
  padding-bottom: 5px;
}

.scale {
  background: linear-gradient(to right, #90C1DE 0 8.33%, #F7591E 8.33% 16.66%, #2AB1A2 16.66% 25%, #FDF147 25% 33.32%, #A0E2CF 33.32% 41.65%, #ED87DD 41.65% 50%, #E26112 50% 58.31%, #72C7C0 58.31% 66.64%, #6FDFAB 66.64% 74.97%, #8793FF 74.97% 83.3%, #E0B076 83.3% 91.63%, #89C781 91.63% 100%);
  height: 70px;
  padding-top: 5px;
}

.settings {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 25%;
  background-color: #3E807A;
  border-top-right-radius: 15px;
  border-bottom: none;
  padding-top: 10px;
  padding-bottom: 10px;
}

.oct_ins_pre {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.prehear_container, .instrument_container, .octave_container {
  margin-bottom: 15px;
}

.octave_container {
  text-align: center;
  line-height: 25px;
}

.octave {
  width: 100px;
}

.login_message {
  margin-left: auto;
  margin-right: auto;
  text-align: center;
}

.hidden {
  display: none;
}

.save_progression {
  width: 90px;
  border: 1px solid black;
  background: none;
  font-size: 15px;
  padding-left: 5px;
  padding-right: 5px;
  padding-bottom: 5px;
}

.new_save_tooltip, .update_tooltip {
  position: absolute;
  background: white;
  opacity: .8;
  border: 1px solid black;
  height: 100px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding-left: 20px;
  padding-right: 20px;
}

.new_save_tooltip input, .update_tooltip input {
  border: 1px solid black;
  font-size: 15px;
  padding-left: 3px;
}

.new_save_tooltip button, .update_tooltip button {
  border: 1px solid black;
  background: none;
  padding: 3px;
}

.back {
    background: none;
    text-decoration: none;
    border: none;
    font-size: 20px;
    margin: 5px;
}

.selected {
    color: white;
    background-color: black;
}

.save_cancel {
  width: 100%;
  display: flex;
  justify-content: space-around;
}

.hidden {
  display: none;
}

.scale:hover, .note:hover {
  opacity: 75%;
}

.insert_chord {
  width: 70px;
  padding: 5px;
  background: none;
  font-size: 18px;
  border: 1px black solid;
}

.insert_chord:disabled {
  border: none;
}

.insert_chord:hover:enabled {
  animation: animation-insert 1s linear infinite;
  background-size: 800% 800%;
  background-image: repeating-linear-gradient(to left, violet, indigo, blue, green, yellow, orange, red);
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
}

.insert_chord:active:enabled {
  color: black;
}

/* scale colorization */
.maj { 
  filter: grayscale(0%);
}

.min {
  filter: grayscale(10%);

}

.maj7 {
  filter: grayscale(20%);

}

.min7 {
  filter: grayscale(30%);

}

.dim {
  filter: grayscale(40%);

}

.dom7 {
  filter: grayscale(50%);

}

.sus2 {
  filter: grayscale(60%);
}

.sus4 {
  filter: grayscale(70%);
}

.aug {
  filter: grayscale(80%);
}

/* chord colorization */

.a {
  background-color: #D41111;
}

.bb {
  background-color: #FFCF5D;
}

.b {
  background-color: #FF8A00;
}

.c {
  background-color: #FFF600;
}

.db {
  background-color: #96D782;
}

.d {
  background-color: #46B523;
}

.eb {
  background-color: #82D2D7;
}

.e {
  background-color: #0CB4D2;
}

.f {
  background-color: #7b62c4;
}

.gb {
  background-color: #DA81E7;
}

.g {
  background-color: #B80CD2;
}

.ab {
  background-color: #FF8282;
}

@keyframes animation-insert {
  0% {
    background-position: right;
  }
  100% {
    background-position: left;
  }
}

@media screen and (max-width: 700px) {

  .player_timeline {
    width: 300px;
  }

  .scrubber_buttons {
    width: 300px;
    justify-content: space-around;
    padding-top: 10px;
  }

  .slide_container {
    margin-left: 10px;
    margin-right: 10px;
  }

  .volume_slider, .speed_slider {
    width: 100px;
  }

  .library_settings {
    flex-direction: column;
    width: 325px;
    height: 800px;
  }

  .settings {
    height: 25%;
    width: 100%;
    flex-direction: row;
    border-top-left-radius: 15px;
  }

  .library {
    margin-top: 30px;
    width: 100%;
  }

  .oct_ins_pre {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    padding-top: 10px;
    margin-left: auto;
    margin-right: auto;
    height: 100%
  }

  .new_save_tooltip, .update_tooltip {
    bottom: initial;
  }

  .scale, .note {
    width: 130px;
    margin-bottom: 40px;
  }

  .note {
    font-size: 40px;
  }

  .scale {
    display: inline-block;
    font-size: 50px;
  }

  .scales {
    columns: 2;
    padding-left: 14px;
    padding-top: 40px;
  }

  .login_message {
    width: 130px;
  }

  .back {
    margin: 15px;
  }
 
  .login_message {
    width: 100px;
  }

  .insert_chord {
    margin-left: auto;
    margin-right: auto;
  }

  .insert_chord:disabled {
    color: lightgray;
    border: 1px solid lightgray;
  }

  .insert_chord:enabled {
    color: black;
  }

  .save_progression {
    margin-left: auto;
    margin-right: auto;
  }
}

@media screen and (min-width: 700px) {
  .settings {
    border-bottom-right-radius: 15px;
  }
}