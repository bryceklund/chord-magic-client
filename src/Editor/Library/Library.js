import React, { Component } from 'react';
import '../Synth.js'
import './Library.css'

class Library extends Component {
    constructor(props) {
        super(props)
        this.state = {
            toggleHidden: false,
            scale: null,
            chord: null,
            octave: null,
            voice: null,
        }
    }

    toggleHidden = () => {
        this.setState({
            toggleHidden: !this.state.toggleHidden
        })
    }

    highlightSelection = (selection) => {
        console.log('highlightSelection running')
        console.log(this.state.chord)
        console.log(selection.textContent)
        console.log(this.state)
        if (selection.textContent === this.state.chord) {
            console.log('highlighting selection')
            selection.classList.add('selected')
            this.forceUpdate()
        }
    }

    setScale = (scale) => {
        this.toggleHidden()
        console.log(this.state.scale)
        this.setState({
            scale: scale.toLowerCase()
        })
    }

    setChord = (chord, target) => {
        this.setState({
            chord
        },  this.highlightSelection(target))
    }

    render() {
        return (
            <section className='library_container'>
            <div className='library_settings'>
                <div className='settings'>
                  <div className='oct_ins_pre'>
                    <div className='prehear_container'>
                      <input className='prehear' id='prehear' type="checkbox" />
                      <label className='prehear_label' htmlFor='prehear'>prehear</label>
                    </div>
                    <div className='instrument_container'>
                      <label className='instrument_label' htmlFor='instrument'>instrument:</label>
                      <select className='instrument' id='instrument'>
                        <option value='sine'>sine</option>
                        <option value='triangle'>triangle</option>
                        <option value='square'>square</option>
                        <option value='saw'>saw</option>
                      </select>
                    </div>
                    <div className='octave_container'>
                      <label htmlFor='octave'>octave</label><br />
                      <div className='octave_div'>
                        <label htmlFor='-2'>-2</label>
                        <input type='radio' name='octave' value='-2' id='-2' />
                      </div>
                      <div className='octave_div'>
                        <label htmlFor='-1'>-1</label>
                        <input type='radio' name='octave' value='-1' id='-1' />
                      </div>
                      <div className='octave_div'>
                        <label htmlFor='0'>0</label>
                        <input type='radio' name='octave' value='0' id='0' />
                      </div>
                      <div className='octave_div'>
                        <label htmlFor='1'>1</label>
                        <input type='radio' name='octave' value='1' id='1' />
                      </div>
                      <div className='octave_div'>
                        <label htmlFor='2'>2</label>
                        <input type='radio' name='octave' value='2' id='2' />
                      </div>
                    </div>
                  </div>
                <p className='login_message'>Login to save your progression!</p>
                <button className='save_progression hidden'>save progression</button>
      
              </div>
              <div className='library'>
                  <button class={`back ${!this.state.toggleHidden ? 'hidden' : ''}`} onClick={() => this.toggleHidden()}>&lt; back to scales</button>
                <ul className={`scales ${this.state.toggleHidden ? 'hidden' : ''}`}>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Maj</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Min</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Maj7</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Min7</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Dim</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Dom7</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Sus2</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Sus4</li>
                    <li className='scale' onClick={(e) => this.setScale(e.target.textContent)}>Aug</li>
                </ul>
                <ul className={`scales chords ${!this.state.toggleHidden ? 'hidden' : ''}`}>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>A</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>Bb</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>B</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>C</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>C#</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>D</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>Eb</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>E</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>F</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>F#</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>G</li>
                    <li className='scale' onClick={(e) => this.setChord(e.target.textContent, e.target)}>Ab</li>
                </ul>
              </div>
      
            </div>
          </section>
      
        );
    }
}

export default Library;