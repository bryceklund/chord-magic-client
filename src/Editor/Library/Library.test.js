import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Library from './Library'
import { exportAllDeclaration } from '@babel/types'

describe(`Library`, () => {
    it(`renders logged in without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter>
                            <Library signedIn={true} />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`renders logged out without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter>
                            <Library signedIn={false} />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`matches its snapshot`, () => {
        const tree = renderer.create(<BrowserRouter>
                                        <Library />
                                    </BrowserRouter>)
                                    .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
