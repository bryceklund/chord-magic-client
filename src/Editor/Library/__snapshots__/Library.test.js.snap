// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Library matches its snapshot 1`] = `
<section
  className="library_container"
>
  <div
    className="library_settings"
  >
    <div
      className="settings"
    >
      <div
        className="oct_ins_pre"
      >
        <div
          className="prehear_container"
        >
          <input
            className="prehear"
            id="prehear"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="prehear_label"
            htmlFor="prehear"
          >
            prehear
          </label>
        </div>
        <div
          className="instrument_container"
        >
          instrument
          <div
            className="wave_icons_container"
          >
            <div
              className="sine_triangle"
            >
              <label>
                <input
                  name="voice"
                  onClick={[Function]}
                  type="radio"
                  value="sine"
                />
                <img
                  className="wave_icon"
                  src="sine.png"
                />
              </label>
              <label>
                <input
                  defaultChecked={true}
                  name="voice"
                  onClick={[Function]}
                  type="radio"
                  value="triangle"
                />
                <img
                  className="wave_icon"
                  src="triangle.png"
                />
              </label>
            </div>
            <div
              className="saw_square"
            >
              <label>
                <input
                  name="voice"
                  onClick={[Function]}
                  type="radio"
                  value="sawtooth"
                />
                <img
                  className="wave_icon"
                  src="sawtooth.png"
                />
              </label>
              <label>
                <input
                  name="voice"
                  onClick={[Function]}
                  type="radio"
                  value="square"
                />
                <img
                  className="wave_icon"
                  src="square.png"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          className="octave_container"
        >
          <label
            htmlFor="octave"
          >
            octave: 
            0
          </label>
          <br />
          <input
            className="octave"
            defaultValue="0"
            max="2"
            min="-2"
            name="octave"
            onChange={[Function]}
            type="range"
          />
        </div>
      </div>
      <button
        className="insert_chord"
        disabled={true}
        onClick={[Function]}
      >
        insert chord
      </button>
      <p
        className="login_message"
      >
        Log in to save your progression!
      </p>
      <div
        className="new_save_tooltip response hidden"
        id="response"
      >
        
      </div>
      <div
        className="update_tooltip hidden"
        id="update_tooltip"
      >
        <button
          className="save"
          onClick={[Function]}
        >
          Save as new
        </button>
        <button
          className="save"
          onClick={[Function]}
        >
          Overwrite existing
        </button>
        <button
          className="save"
          onClick={[Function]}
        >
          Cancel
        </button>
      </div>
      <div
        className="new_save_tooltip hidden"
        id="new_save_tooltip"
      >
        <label
          htmlFor="prog_name"
        >
          Name:
        </label>
        <input
          id="prog_name"
          onChange={[Function]}
          type="text"
        />
        <div
          className="save_cancel"
        >
          <button
            className="save"
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="save"
            onClick={[Function]}
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
    <div
      className="library"
    >
      <button
        className="back hidden"
        onClick={[Function]}
      >
        &lt; back to scales
      </button>
      <ul
        className="scales "
      >
        <li
          className="scale"
          onClick={[Function]}
        >
          Maj
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Min
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Maj7
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Min7
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Dim
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Dom7
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Sus2
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Sus4
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Aug
        </li>
      </ul>
      <ul
        className="scales chords hidden"
      >
        <li
          className="scale"
          onClick={[Function]}
        >
          A
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Bb
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          B
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          C
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Db
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          D
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Eb
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          E
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          F
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Gb
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          G
        </li>
        <li
          className="scale"
          onClick={[Function]}
        >
          Ab
        </li>
      </ul>
    </div>
  </div>
</section>
`;
